inst/rmarkdown/templates/tf/resources/template.tex

```diff
@@ -202,7 +197,7 @@
 $body$
 
 $if(bibliography)$
-\bibliographystyle{$if(biblio-style)$$biblio-style$$else$tfcad$endif$}
+\bibliographystyle{$bst-name$}
```

Comment:
Why not use `biblio-style` here ? Why change to `bst-name` ? 

I think I would 

- Set `biblio-style` in the R function as you did.
- And add a check in the R function at the start to warn if `biblio-style` is already set in YAML metadata by user. Passing it as a variable will override the metadata. 
We do metadata check in pre_knit function for example 
https://github.com/rstudio/rticles/blob/fe7347dba3c4c54e262004ac943ce705c2c6efc2/R/ams_article.R#L45-L54 
or pre_processor
https://github.com/rstudio/rticles/blob/dfca82774dfd9bd9240a6217e2c4a11f582f6399/R/mdpi_article.R#L86-L93)

That would work right ?